src/sql/workbench/services/connection/test/browser/connectionDialogService.test.ts

```diff
@@ -307,11 +307,17 @@ suite('ConnectionDialogService tests', () => {
 		mockWidget.setup(x => x.fillInConnectionInputs(TypeMoq.It.isAny())).returns(() => {
 			called = true;
 		});
+
 		await connectionDialogService.showDialog(mockConnectionManagementService.object, testConnectionParams, connectionProfile);
 		await (connectionDialogService as any).handleFillInConnectionInputs(connectionProfile);
-		let returnedModel = ((connectionDialogService as any)._connectionControllerMap['MSSQL'] as any)._model;
-		assert.strictEqual(returnedModel._groupName, 'testGroup');
-		assert(called);
+
+		setTimeout(() => {
```

Comment:
You aren't waiting for this callback to be called so it will finish the test before any of the code inside is ran. You should use the `done` callback to tell the test to wait until that happens so you're sure that happens.

But also is there a reason you couldn't have handleFillInConnectionInputs return a promise? Doing waits like this generally leads to unstable tests since there's nothing guaranteeing that it will complete before the 200ms is up.